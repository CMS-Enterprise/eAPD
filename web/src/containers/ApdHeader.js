import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import { getAPDCreation, getAPDName, getAPDYearRange } from '../reducers/apd';

const ApdHeader = ({ apdCreated, apdName, year }) => (
  <h1 className="ds-h1 apd--title">
    <span className="ds-h6 ds-u-display--block">
      <strong>Created:</strong> {apdCreated}
    </span>
    {apdName} | FFY {year}
  </h1>
);

ApdHeader.propTypes = {
  apdCreated: PropTypes.string.isRequired,
  apdName: PropTypes.string,
  year: PropTypes.string.isRequired
};

ApdHeader.defaultProps = { apdName: '' };

const mapStateToProps = (state) => ({
  apdCreated: getAPDCreation(state),
  apdName: getAPDName(state),
  year: getAPDYearRange(state)
});

export default connect(mapStateToProps)(ApdHeader);

export { ApdHeader as plain, mapStateToProps };
