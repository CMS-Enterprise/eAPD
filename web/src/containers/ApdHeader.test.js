import { shallow } from 'enzyme';
import React from 'react';

import ApdHeader from './ApdHeader';

describe('APD header', () => {
  it('renders correctly', () => {
    expect(
      shallow(
        <ApdHeader
          apdCreated="creation date"
          apdName="namey name"
          year="ffys"
        />
      )
    ).toMatchSnapshot();
  });
});
