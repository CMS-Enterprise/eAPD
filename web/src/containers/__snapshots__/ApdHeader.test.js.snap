// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APD header renders correctly 1`] = `
<h1
  className="ds-h1 apd--title"
>
  <span
    className="ds-h6 ds-u-display--block"
  >
    <strong>
      Created:
    </strong>
     
    creation date
  </span>
  namey name
   | FFY 
  ffys
</h1>
`;
